### PR TITLE
fix(provider/kuberentes): v1 fix scaleTargetRef apiVersion in HPA

### PR DESCRIPTION
**Pull request by axot**
This will fix https://github.com/spinnaker/spinnaker/issues/2249 , I tested this in kubernetes 1.8.8 and 1.93 in GKE.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/KubernetesUtil.groovy [hunk 1]
```diff
@@ -34,6 +35,8 @@ class KubernetesUtil {
   static String DEPRECATED_SERVER_GROUP_KIND = "ReplicationController"
   static String SERVER_GROUP_KIND = "ReplicaSet"
   static String DEPLOYMENT_KIND = "Deployment"
+  static String SERVER_GROUP_API_VERSION = new ReplicaSet().getApiVersion()
+  static String DEPLOYMENT_API_VERSION = new Deployment().getApiVersion()
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/KubernetesUtil.groovy]: When we bump the underlying client version, do we now risk breaking user's deployments that depended on a certain workload version?

Comment (axot) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/KubernetesUtil.groovy]: Thank you for review, I change the way for api version detection.

Comment (axot) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/KubernetesUtil.groovy]: How about this way?

**Comment by axot** [comment]
@lwander Hi, could you review the new commit please, it gets apiVersion from exists development/replicaset now.

**Comment by lwander** [comment]
Thanks!

**Pull request merged**